sql/core/src/main/scala/org/apache/spark/sql/catalyst/SQLBuilder.scala

```diff
@@ -356,8 +372,14 @@ class SQLBuilder(logicalPlan: LogicalPlan) extends Logging {
       }
     }
 
+    val broadcastHint = project match {
+      case p @ Project(projectList, Hint("BROADCAST", tables, child)) =>
+        if (tables.nonEmpty) s"/*+ MAPJOIN(${tables.mkString(", ")}) */" else ""
+      case _ => ""
+    }
     build(
       "SELECT",
+      broadcastHint,
```

Comment:
This is in `groupingSetToSQL`. This looks weird. How about the others? e.g., `windowToSQL`?